src/main/drivers/adc.h

```diff
@@ -18,6 +18,7 @@
 #pragma once
 
 #include "drivers/io_types.h"
+#include "pg/adc.h"
```

Comment:
This is not really needed in here. Should this go into `adc.c`? Not sure if promoting indirect includes is a good thing.

Reply:
@ledvinap: Your reaction is not quite clear. Can you elaborate?

Reply:
(Sorry, not that sure now) 
Code exporting `adcConfig_t*` in it's interface should also provide its declaration. pg/adc.h is quite minimalistic, using it is reasonable. But it will pull "config/parameter_group.h", which is not that good.

Maybe just use forward declaration of `adcConfig_t` (`struct adcConfig_s`)

Reply:
Good suggestion. Fixed using forward declaration. Now `pg/adc.h` is not included in any header files.

Reply:
I know we are slowly going out on a limb here, but how about sticking the pg struct into a `_struct` file in the `parameter_groups` directory and including that in the driver and in the actual pg declaration? This way it's very clear that the struct is subject to the restrictions that come with being part of a pg, yet drivers are prevented from using the pg directly.

Reply:
I know we are slowly going out on a limb here, but how about sticking the pg struct into a `_struct` file in the `parameter_groups` directory and including that in the driver and in the actual pg declaration? This way it's very clear that the struct is subject to the restrictions that come with being part of a pg, yet drivers are prevented from using the pg directly.

Reply:
I know we are slowly going out on a limb here, but how about sticking the pg struct into a `_struct` file in the `parameter_groups` directory and including that in the driver and in the actual pg declaration? This way it's very clear that the struct is subject to the restrictions that come with being part of a pg, yet drivers are prevented from using the pg directly.

Reply:
@mikeller : Yes, it's valid point, but IMO not worth the effort. This is C, macros and includes are leaking everywhere (and it would take much bigger effort to fix).
With forward declaration ADC user sees opaque configuration type and doesn't need PG.
Only driver implementation will pull in PG, that's only few files, IMO it's acceptable penalty.